core/ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/debug/DebuggerManager.java

```diff
@@ -40,25 +40,22 @@ protected DebuggerManager(AppContext appContext) {
 
     /**
      * Register new debugger for the specified project type ID.
+     * TODO: don't link debugger with project.
      *
      * @param projectTypeId
      * @param debugger
      */
     public void registeredDebugger(String projectTypeId, Debugger debugger) {
         debuggers.put(projectTypeId, debugger);
+        setDebugger(debugger);
```

Comment:
I would like to keep as is. Because right now I am working on refactoring all classed related to debug. So, DebuggerManager will be change significantly